packages/fxa-settings/src/lib/error-utils.ts

```diff
@@ -0,0 +1,175 @@
+      retryAfterLocalized:
+        (graphQLError?.extensions.retryAfterLocalized as string) || undefined,
+    };
+    return { error: uiError as HandledError };
+  }
+
+  // if not a graphQLError or if no localizable message available for error
+  return { error: AuthUiErrors.UNEXPECTED_ERROR as HandledError };
+};
+
+/**
+ * Utility function to retrieve the localized auth client error message
+ * - works for throttling errors that include a localized retry after value
+ * - returns an unexpected error if the error is unknown or does not have a localized message
+ * @param ftlMsgResolver
+ * @param err is an AuthClient error
+ * @returns
+ */
+export const getLocalizedErrorMessage = (
+  ftlMsgResolver: FtlMsgResolver,
+  error: AuthUiError | HandledError
+) => {
+  if (error.errno) {
+    if (AuthUiErrorNos[error.errno]) {
+      if (
+        error.retryAfterLocalized &&
+        error.errno === AuthUiErrors.THROTTLED.errno
+      ) {
+        // For throttling errors where a localized retry after value is provided
+        return ftlMsgResolver.getMsg(
+          getErrorFtlId(error),
+          AuthUiErrorNos[error.errno].message,
+          { retryAfter: error.retryAfterLocalized }
+        );
+      } else if (error.errno === AuthUiErrors.THROTTLED.errno) {
+        // For throttling errors where a localized retry after value is not available
+        return ftlMsgResolver.getMsg(
+          'auth-error-114-generic',
+          AuthUiErrorNos[114].message
+        );
+      } else {
+        // for all other recognized auth UI errors
+        return ftlMsgResolver.getMsg(
+          getErrorFtlId(error),
+          AuthUiErrorNos[error.errno].message
+        );
+      }
+    }
+    const oAuthError = Object.values(OAUTH_ERRORS).find(
+      (oAuthErr) => error.errno === oAuthErr.errno
+    );
+    if (oAuthError) {
+      return ftlMsgResolver.getMsg(
+        getErrorFtlId(oAuthError),
+        oAuthError.message
+      );
+    }
+  }
+
+  const unexpectedError = AuthUiErrors.UNEXPECTED_ERROR;
+  return ftlMsgResolver.getMsg(
+    getErrorFtlId(unexpectedError),
+    unexpectedError.message
+  );
+};
+
+/*
+  TODO in FXA-9502: account for potential errno overlap between auth-errors
+  and oauth errors. Checking the auth-error array currently happens first.
+**/
+export const getErrorFtlId = (err: { errno?: number; message?: string }) => {
+  if (err.errno) {
+    if (err.errno in AuthUiErrorNos) {
+      const error = AuthUiErrorNos[err.errno];
+      return `auth-error-${err.errno}${
+        error.version ? '-' + error.version : ''
+      }`;
+    }
+
+    const oAuthError = Object.values(OAUTH_ERRORS).find(
+      (oAuthErr) => err.errno === oAuthErr.errno
+    );
+    if (oAuthError) {
+      return `oauth-error-${err.errno}${
+        oAuthError.version ? '-' + oAuthError.version : ''
+      }`;
+    }
+  }
+  // If the error isn't found, return an empty string FTL ID and log to Sentry.
+  const logMessage = err.errno
```

Comment:
I think errors that are commented out are obsolete and shouldn't be thrown/displayed at all, right (which would be unexpected and should probably be logged)? Or are there any you know of that we've commented out because of l10n? 🤔 This was existing code FWIW (but still a good question to ask).